server/src/challenge/challenge.service.ts

```diff
@@ -132,4 +139,39 @@ export class ChallengeService {
 
     return eventTracker;
   }
+
+  /** Check if the current event can return rewards */
+  async checkForReward(user: User, eventBase: EventBase, eventTracker: EventTracker) {
```

Comment:
You might actually want to return Promise<EventReward | null> from this function and return a reward or null depending on what you do in this function. That way you can use the reward you picked out in this function (if it's not null) and call emitUpdateUserData from the gateway.